File: 0014-longest-common-prefix/0014-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    
    var indexs=strs[0].length;
    
    var sol=strs[0];
strs.map(str=>{
        var i=0;
        while(i<indexs&&i<str.length){
            if(str.charAt(i)===sol.charAt(i)){
                i++;
            }
            else break;
        }
        indexs=i;
    });

  return sol.slice(0,indexs);  
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        lng=len(strs[0])
        for i in range(1,len(strs),1):
            lng=min(lng,len(strs[i]))
            for j in range(min(lng,len(strs[i]))):
                if strs[0][j]!=strs[i][j]:
                    lng=j
                    break
        return strs[0][0:lng]


        